src/Tools/GetDocumentInsider/src/ProgramBase.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+using Microsoft.Extensions.ApiDescription.Tool.Commands;
+using Microsoft.Extensions.CommandLineUtils;
+using Microsoft.Extensions.Tools.Internal;
+
+namespace Microsoft.Extensions.ApiDescription.Tool
+{
+    internal abstract class ProgramBase
+    {
+        private readonly IConsole _console;
+        private readonly IReporter _reporter;
+
+        public ProgramBase(IConsole console)
+        {
+            _console = console ?? throw new ArgumentNullException(nameof(console));
+            _reporter = new ConsoleReporter(_console, verbose: false, quiet: false);
```

Comment:
Does it make more sense to treat quiet/verbose similar to debug and pre-parse those arguments -> Create a reporter at the top level -> flow to any command interested? I ask because you create a reporter here in `ProgramBase`, the `CommandBase` constructor and in the `CommandBase` configure. Seems like a whole lot of maintenance for a reporter.

Reply:
`--debug` is "special" because it's supported only as the first command-line argument, not understood by any `CommandOption`, and generally hidden. The others are real options that can appear anywhere in the command line. Besides that, this is the pattern we've used elsewhere.

BTW the truly-extra `Reporter` instances are created for `Exception` handling in `Program` classes whether or not the command line has been parsed. The `--quiet` and `--verbose` options don't matter when writing errors.